File: Leetcode/1.cpp
/**
 * 1. 两数之和
 * link：https://leetcode-cn.com/problems/two-sum/
 */

class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        unordered_map<int, int> record;

        for (int i = 0; i < nums.size(); i++)
        {
            if (record.find(target - nums[i]) != record.end())
                return {record[target - nums[i]], i};
            record[nums[i]] = i;
        }

        return {};
    }
};

File: Leetcode/1.py
# 1. 两数之和
# link：https://leetcode-cn.com/problems/two-sum/

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashset = dict()
        for idx, num in enumerate(nums):
            if (target - num) in hashset:
                return [hashset.get(target - num), idx]
            hashset[num] = idx
        return []
